bezier_library/src/bezier_library.cpp

```diff
@@ -1112,80 +1112,102 @@ bool Bezier::generateTrajectory(
   return true;
 }
 
-void Bezier::displayNormal(
+void Bezier::RVIZdisplayNormals(
     const std::vector<Eigen::Affine3d, Eigen::aligned_allocator<Eigen::Affine3d> > &way_points_vector,
     const std::vector<bool> points_color_viz,
-    const ros::Publisher &normal_publisher)
+    const std::string name,
+    const ros::Publisher &normal_publisher,
+    const std::string where)
 {
-  if (way_points_vector.size() != points_color_viz.size())
-    ROS_ERROR_STREAM("Bezier::displayNormal: Path vector and bool vector have different sizes");
 
-  // Delete old markers
-  visualization_msgs::MarkerArray markers;
-  for (unsigned int k = 0; k < number_of_normal_markers_published_; k++)
+  if (way_points_vector.size() != points_color_viz.size())
   {
-    visualization_msgs::Marker marker;
-    marker.header.frame_id = "/base_link";
-    marker.header.stamp = ros::Time::now();
-    marker.ns = "basic_shapes";
-    marker.id = k; // It doesn't matter if the marker was published or not (depending on points_color_viz[k])
-    marker.type = visualization_msgs::Marker::ARROW;
-    marker.action = visualization_msgs::Marker::DELETE;
-    markers.markers.push_back(marker);
+    ROS_ERROR_STREAM("Bezier::RVIZdisplayNormals: Path vector and bool vector have different sizes");
+    return;
   }
-  normal_publisher.publish(markers);
 
-  number_of_normal_markers_published_ = way_points_vector.size(); // Store number of markers drawn
-  for (unsigned int k = 0; k < way_points_vector.size(); k++)
+  visual_tools_.reset(new rviz_visual_tools::RvizVisualTools("base_link",name));
```

Comment:
This must be done only one time in the Bezier constructor; `base_link` must be a parameter of the constructor. name too.